### PR TITLE
Added Arch linux package generation.

### PR DESCRIPTION
**Pull request by excitoon**
I would like to have a way to quickly install built code to my Arch linux system.
I think it is a good idea to generate a package along with the binaries. Also this is a step to making automatic releases to GitHub.
Maybe @kmeaw would make some suggestions to make it useful for more people.
For example, allow to generate package only if STATIC options are ON or to generate dependencies list according to CMake flag set?

### CODE REVIEW

**Comment by proller** [comment]
It is possible to place new files to utils/package/arch/ ?

**Comment by excitoon** [comment]
Okay!

File: arch/PKGBUILD.in [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    cp $SRC/dbms/programs/server/config.xml $SRC/dbms/programs/server/users.xml $pkgdir/etc/clickhouse-server/
+    cp $BIN/dbms/programs/clickhouse $pkgdir/usr/bin/clickhouse-client
+    patchelf --remove-rpath $pkgdir/usr/bin/clickhouse-client
+    patchelf --replace-needed libz.so.1 libz-ng.so.1 $pkgdir/usr/bin/clickhouse-client
+    cp $SRC/dbms/programs/client/clickhouse-client.xml $pkgdir/etc/clickhouse-client/config.xml
+    compiler="libclickhouse-compiler.so"
+    if ! pacman -Q clang | grep '^clang 7'; then
+        compiler=""
+    fi
+    sed -e 's:/opt/clickhouse:/var/lib/clickhouse:g' -i $pkgdir/etc/clickhouse-server/config.xml
```
Comment (proller) [inline arch/PKGBUILD.in]: We cnahge /opt to /var/lib long time ago

File: arch/PKGBUILD.in [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    sed -e '/listen_host/s%::<%::1<%' -i $pkgdir/etc/clickhouse-server/config.xml
```
Comment (proller) [inline arch/PKGBUILD.in]: listen is ::1 by default now

File: arch/PKGBUILD.in [hunk 3]
```diff
@@ -0,0 +1,29 @@
+pkgname=clickhouse
+pkgver=${VERSION_STRING}
+pkgrel=1
+pkgdesc='An open-source column-oriented database management system that allows generating analytical data reports in real time'
+arch=('x86_64')
+url='https://clickhouse.yandex/'
+license=('Apache')
+
+package() {
+    # This code was requisited from kmeow@ https://aur.archlinux.org/packages/clickhouse/ .
```
Comment (alexey-milovidov) [inline arch/PKGBUILD.in]: His name is `kmeaw`, not `kmeow`.

**Pull request merged**